tests/frontend/test_ops_decompositions.py

```diff
@@ -275,6 +275,30 @@ def test_decomposition(self, hbar, tol):
         assert np.allclose(ratio, np.ones([n, n]), atol=tol, rtol=0)
 
 
+
+    def test_decomposition_interferometer_with_zero(self, hbar, tol):
+        """Test that an graph is correctly decomposed when the interferometer
+        has one zero somewhere is the unitary matrix, which is the case for the
+        Adjacency matrix below"""
+        n = 6
+        prog = sf.Program(n)
+
+        A = np.array([
+        [0, 1, 0, 0, 1, 1],
+        [1, 0, 1, 0, 1, 1],
+        [0, 1, 0, 1, 1, 0],
+        [0, 0, 1, 0, 1, 0],
+        [1, 1, 1, 1, 0, 1],
+        [1, 1, 0, 0, 1, 0],
+        ]
+        )
+        sq, U = dec.graph_embed(A)
+        assert not np.allclose(U, np.identity(n))
+        G = ops.GraphEmbed(A)
+        cmds = G.decompose(prog.register)
+
+        assert str(cmds[-1].op)[:4] == "Interferometer"[:4]
```

Comment:
This assert didn't make much sense to me (checked that the first four letters of the string-converted op were "Inte"), so I updated it to do something which makes the intention more clear